h/views/api_users.py

```diff
@@ -89,6 +90,8 @@ def _request_client(request):
         raise ClientUnauthorized()
     if client.secret is None:  # client is not confidential
         raise ClientUnauthorized()
+    if client.grant_type != GrantType.client_credentials:  # client not allowed to create users
```

Comment:
You raise a good point. I think we should revisit this when we implement the edit user API endpoint which will have to use the same authorization logic.